rs/pocket_ic_server/src/main.rs

```diff
@@ -222,15 +224,18 @@ async fn start(runtime: Arc<Runtime>) {
             tokio::time::sleep(Duration::from_secs(1)).await;
         }
 
-        debug!("The PocketIC server will terminate");
-
-        shutdown_handle.shutdown();
-
-        if let Some(port_file_path) = port_file_path_clone {
-            // Clean up port file.
-            let _ = std::fs::remove_file(port_file_path);
-        }
+        terminate(app_state, shutdown_handle, port_file_path_clone).await;
+    });
+    // Register a signal handler.
+    let (tx, rx) = channel();
+    let shutdown_handle = handle.clone();
+    let port_file_path_clone = port_file_path.clone();
+    tokio::spawn(async move {
+        rx.recv().expect("Did not receive a signal.");
```

Comment:
Good point! I couldn't use a tokio channel since `ctrlc::set_handler` takes a (sync) closure. What I have done now is to use `try_recv` and wait for 1s if the channel is empty: https://github.com/dfinity/ic/pull/2043/commits/3ea45326be9e8ff15d55b4191ba13ba2f9db7b1e. Does this make sense to you? Also should I reduce the polling period for the sake of dfx stopping quickly?

Reply:
Tokio channels have a blocking_send function for this purpose.

Reply:
Next try: https://github.com/dfinity/ic/pull/2043/commits/1cdd726bf2d22a0af97ed17ff7d248ba12f5ecc2